### PR TITLE
[CN-220] Improve operator status for client

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Fixed race condition between `initHzClient` and `shutdown`, where if `StartNewClientWithConfig` takes too much time reconcile function gets blocked for a very long time.
- Improved Hazelcast status messages about client's connection to the cluster. 

TODO:
- [ ] Operator still cannot understand when a connected client gets disconnected from the cluster. We need to use the functions in the client introduced in this commit https://github.com/hazelcast/hazelcast-go-client/commit/6845efd6ac3d6826c75e23ed75705b74c48a21aa.

### CODE REVIEW

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (hakanmemisoglu) [inline api/v1alpha1/zz_generated.deepcopy.go:1]: I think you can delete this line.

**Comment by hasancelik** [comment]
> TODO: [] Operator still cannot understand when a connected client gets disconnected from the cluster. We need to use the functions in the client introduced in this commit https://github.com/hazelcast/hazelcast-go-client/commit/6845efd6ac3d6826c75e23ed75705b74c48a21aa.

It is satisfied with https://github.com/hazelcast/hazelcast-platform-operator/pull/180, right?

**Comment by mtyazici** [comment]
@hasancelik, I don't think it solves the problem. In that PR we do not check if the client is connected to the cluster but we only check member connection.

File: controllers/hazelcast/hazelcast_controller.go [hunk 2]
```diff
@@ -226,6 +228,24 @@ func getHazelcastCRName(pod *corev1.Pod) (string, bool) {
 	}
 }
 
+func clientConnectionMessage(req ctrl.Request) string {
+	c, ok := GetClient(req.NamespacedName)
+	if !ok {
+		return "Operator failed to create connection to cluster . Some features might be unavailable."
```
Comment (hakanmemisoglu) [inline controllers/hazelcast/hazelcast_controller.go:234]:
```suggestion
		return "Operator failed to create connection to cluster , some features might be unavailable."
```

Comment (hakanmemisoglu) [inline controllers/hazelcast/hazelcast_controller.go:234]:
```suggestion
		return "Operator failed to create connection to cluster, some features might be unavailable."
```

**Review by hakanmemisoglu — APPROVED** [review]

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**